src/internal/SchemaBase.ts

```diff
@@ -0,0 +1,159 @@
+import * as SC from '../SchemaExt'
+import { Refinement } from 'fp-ts/Refinement'
+import * as S2 from './Schemable2'
+import { identity } from 'fp-ts/function'
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type Any = any
+
+const URI = 'SchemaExt'
+
+type URI = typeof URI
+
+declare module 'fp-ts/lib/HKT' {
+  interface URItoKind2<E, A> {
+    readonly SchemaExt: SC.SchemaExt<E, A>
+  }
+}
+
+/**
+ * @since 1.0.0
+ * @category Constructors
+ */
+export const Literal: S2.Schemable2<URI>['literal'] = (...values) =>
+  SC.make(_ => _.literal(...values))
+
+/**
+ * @since 1.0.0
+ * @category Primitives
+ */
+export const String: S2.Schemable2<URI>['string'] = SC.make(_ => _.string)
+
+/**
+ * @since 1.0.0
+ * @category Primitives
+ */
+export const Number: S2.Schemable2<URI>['number'] = SC.make(_ => _.number)
+
+/**
+ * @since 1.0.0
+ * @category Primitives
+ */
+export const Boolean: S2.Schemable2<URI>['boolean'] = SC.make(_ => _.boolean)
+
+/**
+ * @since 1.0.0
+ * @category Primitives
+ */
+export const UnknownArray: S2.WithUnknownContainers2<URI>['UnknownArray'] = SC.make(
+  _ => _.UnknownArray
+)
+
+/**
+ * @since 1.0.0
+ * @category Primitives
+ */
+export const UnknownRecord: S2.WithUnknownContainers2<URI>['UnknownRecord'] = SC.make(
+  _ => _.UnknownRecord
+)
+
+/**
+ * @since 1.0.0
+ * @category Combinators
+ */
+export const Refine =
+  <A, B extends A>(refinement: Refinement<A, B>, id: string) =>
+  <O>(from: SC.SchemaExt<O, A>): SC.SchemaExt<O, B> =>
+    SC.make(_ => _.refine<A, B>(refinement, id)<O>(from(_)))
+
+/**
+ * @since 1.0.0
+ * @category Combinators
+ */
+export const Nullable: S2.Schemable2<URI>['nullable'] = or =>
+  SC.make(_ => _.nullable(or(_)))
+
+/**
+ * @since 1.0.0
+ * @category Combinators
+ */
+export const Struct: S2.Schemable2<URI>['struct'] = props =>
+  SC.make(_ => {
+    const out = {} as Any
+    for (const [key, schemaKey] of Object.entries(props)) {
+      out[key] = schemaKey(_)
+    }
+    return _.struct(out) as Any
+  })
+
+// TODO @jacob-alford 2022-10-20: add Partial
+// Not quite sure how to do this yet, since undefined is not part of Schemable
```

Comment:
```suggestion
/**
 * @since 1.0.0
 * @category Combinators
 */
export const Partial: S2.Schemable2<URI>['partial'] = props =>
  SC.make(_ => {
    const out = {} as Any
    for (const [key, schemaKey] of Object.entries(props)) {
      out[key] = schemaKey(_)
    }
    return _.partial(out) as Any
  })
```

ie same as `Struct` above, just `_.partial` instead

Reply:
Ooooh, right.  Thanks!

Reply:
:heavy_check_mark: